haskell/src/Lorentz/Contracts/Stablecoin.hs

```diff
@@ -94,46 +94,45 @@ type ParameterC param =
 --- Storage
 ------------------------------------------------------------------
 
-type Ledger = BigMap Address Natural
+type OperatorsInner = BigMap (Address, Address) ()
 
-type Operators = BigMap (Address, Address) ()
+type LedgerInner = Map Address Natural
+type Ledger = "ledger" :! (BigMap Address Natural)
 
-type MintingAllowances = BigMap Address Natural
+type Operators = "operators" :! OperatorsInner
+type TotalSupply = "total_supply" :! Natural
+type IsPaused = "paused" :! Bool
 
-type TotalSupply = Natural
+type MintingAllowancesInner = Map Address Natural
+type MintingAllowances = "minting_allowances" :! (BigMap Address Natural)
 
-type IsPaused = Bool
+type RolesInner = (("master_minter" :! Address, "owner" :! Address)
+             , ("pauser" :! Address, "pending_owner_address" :! Maybe Address))
 
--- We enforce specific storage structure in favor of what ligo generates
-type Storage
-  = ( ( ( Ledger
-        , MintingAllowances
-        )
-      , ( Operators
-        , IsPaused
-        )
-      )
-    , ( ( PermissionsDescriptor
-        , Roles
-        )
-      , ( TokenMetadata
-        , TotalSupply
-        )))
+type Roles = "roles" :! RolesInner
 
-pattern StorageLedger :: Ledger -> Storage
-pattern StorageLedger ledger <- (((ledger, _), _), _)
+type Metadata = "token_metadata" :! TokenMetadata
 
-pattern StorageMinters :: Map Address Natural -> Storage
-pattern StorageMinters minters <- (((_, BigMap minters), _), _)
+type SafelistContract = "safelist_contract" :! (Maybe Address)
 
-pattern StorageOperators :: Operators -> Storage
-pattern StorageOperators operators <- ((_, (operators, _)), _)
+type Storage =
```

Comment:
Nice, once we done with this PR, you should mention this refactoring in commits

Reply:
@awkure Sadly, this commit was lost in a lot of squashed rebases I had to do to keep up with #31. Hope you don't mind.